L1Trigger/L1CaloTrigger/plugins/Phase2L1CaloJetEmulator.cc

```diff
@@ -145,21 +319,42 @@ void Phase2L1CaloJetEmulator::produce(edm::Event& iEvent, const edm::EventSetup&
     if (eta > 1.479)
       ieta = ieta - 4;
     int iphi = it->id().iPhi();
+    float hgcal_etEm = it->etEm();
+    float hgcal_etHad = it->etHad();
+    if (abs(eta) <= 1.8) {
+      hgcal_etEm = it->etEm() - all_nvtx_to_PU_sub_funcs["hgcalEM"]["er1p4to1p8"].Eval(EstimatedNvtx);
+      hgcal_etHad = it->etHad() - all_nvtx_to_PU_sub_funcs["hgcalHad"]["er1p4to1p8"].Eval(EstimatedNvtx);
+    }
+    if (abs(eta) <= 2.1 && abs(eta) > 1.8) {
+      hgcal_etEm = it->etEm() - all_nvtx_to_PU_sub_funcs["hgcalEM"]["er1p8to2p1"].Eval(EstimatedNvtx);
+      hgcal_etHad = it->etHad() - all_nvtx_to_PU_sub_funcs["hgcalHad"]["er1p8to2p1"].Eval(EstimatedNvtx);
+    }
+    if (abs(eta) <= 2.4 && abs(eta) > 2.1) {
+      hgcal_etEm = it->etEm() - all_nvtx_to_PU_sub_funcs["hgcalEM"]["er2p1to2p4"].Eval(EstimatedNvtx);
+      hgcal_etHad = it->etHad() - all_nvtx_to_PU_sub_funcs["hgcalHad"]["er2p1to2p4"].Eval(EstimatedNvtx);
+    }
+    if (abs(eta) <= 2.7 && abs(eta) > 2.4) {
+      hgcal_etEm = it->etEm() - all_nvtx_to_PU_sub_funcs["hgcalEM"]["er2p4to2p7"].Eval(EstimatedNvtx);
+      hgcal_etHad = it->etHad() - all_nvtx_to_PU_sub_funcs["hgcalHad"]["er2p4to2p7"].Eval(EstimatedNvtx);
+    }
+    if (abs(eta) <= 3.1 && abs(eta) > 2.7) {
+      hgcal_etEm = it->etEm() - all_nvtx_to_PU_sub_funcs["hgcalEM"]["er2p7to3p1"].Eval(EstimatedNvtx);
+      hgcal_etHad = it->etHad() - all_nvtx_to_PU_sub_funcs["hgcalHad"]["er2p7to3p1"].Eval(EstimatedNvtx);
+    }
```

Comment:
The only thing that changes between these cases is the string key, the math is the same in each case. It may be better to just determine the key, and just write the math later once.

I think this may be a little less redundant:
```c++
    float hgcal_etEm = it->etEm();
    float hgcal_etHad = it->etHad();
    std::string etaKey = "";
    if (abs(eta) <= 1.8) etaKey = "er1p4to1p8";
    else if (abs(eta) <= 2.1 && abs(eta) > 1.8) etaKey = "er1p8to2p1";
    else if (abs(eta) <= 2.4 && abs(eta) > 2.1) etaKey = "er2p1to2p4";
    else if (abs(eta) <= 2.7 && abs(eta) > 2.4) etaKey = "er2p4to2p7";
    else if (abs(eta) <= 3.1 && abs(eta) > 2.7) etaKey = "er2p7to3p1";
    
    if (etaKey != "") {
      hgcal_etEm = it->etEm() - all_nvtx_to_PU_sub_funcs["hgcalEM"][etaKey].Eval(EstimatedNvtx);
      hgcal_etHad = it->etHad() - all_nvtx_to_PU_sub_funcs["hgcalHad"][etaKey].Eval(EstimatedNvtx);
    }
```

Reply:
Done